src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,91 @@
+package Duke;
+
+import java.util.Objects;
+import java.util.ArrayList;
+
+import static Duke.TaskList.*;
+import static Duke.Ui.*;
+import static Duke.InputCheckingPackage.InputChecking.*;
+
+/**
+ * Duke is a Command Line Application that helps user keep track of things to do,
+ * and deadlines ar events they might have
+ */
+
+public class Duke {
+
+    public static ArrayList<Task> tasks = new ArrayList<>();
+    private static String filePath = "data/tasklist.txt";
+    private Ui ui;
+
+    /**
+     * Initialises Duke application and reads for exsiting tasks that
+     * are saved in a text file
+     * @param filePath path of the text file that contains previous task information
+     */
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        
+        try {
+            Storage.readFile(filePath, tasks);
+        } catch (java.io.FileNotFoundException e) {
+            ui.showLoadingError();
+        }
+    }
+
+    /**
+     * Runs the application and reads the user input and carries out the related
+     * commands until the user chooses to exit the application
+     */
+    public static void run() {
```

Comment:
this method has room for refactoring, it is currently quite hard to trace the code.
consider applying abstraction, refer to addressbook3 if you wish